Facing crash while creating Power Cloud Connections

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_pi_cloud_connection

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
	  resource "ibm_pi_cloud_connection" "cloud_connection" {
		pi_cloud_instance_id				= data.ibm_resource_instance.service.guid
		pi_cloud_connection_name			= var.name
		pi_cloud_connection_speed			= var.speed
		pi_cloud_connection_networks		= [ibm_pi_network.network1.network_id]
	  }

```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->
```

Stack trace from the terraform-provider-ibm_v1.38.0.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x8 pc=0x29b76e9]

goroutine 93 [running]:
github.com/IBM-Cloud/terraform-provider-ibm/ibm.resourceIBMPICloudConnectionCreate(0x38f0798, 0xc000d61ec0, 0xc000d77480, 0x334c1a0, 0xc001442800, 0xc00064b130, 0x1a91cb2, 0xc000fb9e00)
        github.com/IBM-Cloud/terraform-provider-ibm/ibm/resource_ibm_pi_cloud_connection.go:212 +0x949
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000c89c00, 0x38f0728, 0xc0003b8240, 0xc000d77480, 0x334c1a0, 0xc001442800, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:330 +0x187
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000c89c00, 0x38f0728, 0xc0003b8240, 0xc0001a0700, 0xc000fb9e00, 0x334c1a0, 0xc001442800, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00078d500, 0x38f0728, 0xc0003b8240, 0xc001448be0, 0xc0003b8240, 0x3268020, 0xc000637200)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000fb9320, 0x38f07d0, 0xc0003b8240, 0xc0001a04d0, 0xc000fb9320, 0xc0006372f0, 0xc000ff5ba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb6
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x3268020, 0xc000fb9320, 0x38f07d0, 0xc0006372f0, 0xc000d61800, 0x0, 0x38f07d0, 0xc0006372f0, 0xc00092ea00, 0x678)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x222
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001ce8c0, 0x390b238, 0xc0004a2900, 0xc00052eb00, 0xc000d37530, 0x482a980, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0001ce8c0, 0x390b238, 0xc0004a2900, 0xc00052eb00, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000780280, 0xc0001ce8c0, 0x390b238, 0xc0004a2900, 0xc00052eb00)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xb2
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-ibm_v1.38.0.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```
### Expected Behavior

<!--- What should have happened? --->
Cloud connection should be created without the crash.

### Actual Behavior

<!--- What actually happened? --->
`terraform apply` crashed with given stack trace.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

Thanks @surajbharadwaj for reporting the issue.

--- a/ibm/service/power/resource_ibm_pi_cloud_connection.go
+++ b/ibm/service/power/resource_ibm_pi_cloud_connection.go
@@ -212,7 +212,7 @@ func resourceIBMPICloudConnectionCreate(ctx context.Context, d *schema.ResourceD
 	if cloudConnection != nil {
 		d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, *cloudConnection.CloudConnectionID))
 	} else if cloudConnectionJob != nil {
-		d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, *cloudConnection.CloudConnectionID))
+		d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, *cloudConnectionJob.CloudConnectionID))
 
 		jobID := *cloudConnectionJob.JobRef.ID
 

